components/reddit/sources/new-links-by-user/new-links-by-user.js

```diff
@@ -0,0 +1,78 @@
+const common = require("../common");
+const { reddit } = common.props;
+
+module.exports = {
+  ...common,
+  key: "new-links-by-user",
+  name: "New links by user",
+  description: "Emits an event each time a user posts a new link.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    username: { propDefinition: [common.props.reddit, "username"] },
+    numberOfParents: {
+      type: "integer",
+      label: "Number of parents",
+      description:
+        "The emitted events will contain the new comment plus the parents of said comment up to the number indicated in this property.",
+      default: 2,
+      min: 2,
+      max: 10
+    },
+    timeFilter: { propDefinition: [reddit, "timeFilter"] },
+    includeSubredditDetails: {
+      propDefinition: [reddit, "includeSubredditDetails"],
+    },
+  },
+  hooks: {
+    async deploy() {
+      // Emits sample events on the first run during deploy.
+      var redditLinks = await this.reddit.getNewUserLinks(
+        null,
+        this.username,
+        this.numberOfParents,
+        this.timeFilter,
+        this.includeSubredditDetails,
+        10
+      );
+      const { children: links = [] } = redditLinks.data;
+      if (links.length === 0) {
+        console.log("No data available, skipping itieration");
+        return;
+      }      
+      const { name: before = this.db.get("before") } = links[0].data;
+      this.db.set("before", before);
+      links.reverse().forEach(this.emitRedditEvent);
+    },    
+  },
+  methods: {
+    ...common.methods,
+    generateEventMetadata(redditEvent) {
+      return {
+        id: redditEvent.data.name,
+        summary: redditEvent.data.title,
+        ts: redditEvent.data.created,
+      };
+    },
+  },
+  async run() {
+    do {
+      const redditLinks = await this.reddit.getNewUserLinks(
+        this.db.get("before"),
+        this.username,
+        this.numberOfParents,
+        this.timeFilter,
+        this.includeSubredditDetails
+      );
+      const { children: links = [] } = redditLinks.data;
+      if (links.length === 0) {
+        console.log("No data available, skipping itieration");
+        break;
+      }      
+      const { name: before = this.db.get("before") } = links[0].data;
+      this.db.set("before", before);
+      links.reverse().forEach(this.emitRedditEvent);
+    } while (redditLinks);
```

Comment:
`redditLinks` needs to be declared outside the loop (see other related comment in the new comments by user source)

Reply:
Yep, fixed.